addons/web_editor/static/src/scss/web_editor.variables.scss

```diff
@@ -630,6 +630,40 @@ $o-color-extras-nesting-selector: '&, .o_colored_level &';
     @return $color1;
 }
 
+// Given a primary color (and eventually a secondary one), the function returns
+// a basic odoo palette in sass-map format. The palette will be generated using
+// the safest readability values possible.
+@function o-make-palette($-primary, $-secondary: null, $-overrides-map: null) {
+    $-o-color-2: $-secondary or increase-contrast(desaturate(mix(complement($-primary), #FFFFFF, 80%), 20%), $-primary);
+
+    $-palette: (
+        'o-color-1': $-primary,
+        'o-color-2': $-o-color-2,
+        'o-color-3': change-color(#F5F0F0, $hue: hue($-primary), $saturation: min(saturation($-primary), saturation(#F5F0F0))),
+        'o-color-4': #FFFFFF,
+        'o-color-5': change-color(#2e1414, $hue: hue($-primary), $saturation: min(saturation($-primary), saturation(#2e1414))),
```

Comment:
note: @qsm-odoo , I clamp the saturation to avoid that these colors could more saturated that primary ... it's non an "**IF**" 😂

Reply:
A hidden if :grimacing: :smile: